src/deploy/deployer/deployer.ts

```diff
@@ -0,0 +1,48 @@
+import * as hre from "hardhat";
+import { HardhatRuntimeEnvironment } from "hardhat/types";
+import { TDeployArgs, TProxyKind } from "../missions/types";
+
+export class Deployer {
+  hre : HardhatRuntimeEnvironment;
+
+  constructor () {
+    this.hre = hre;
+  }
+
+  async deployProxy ({
+    contractName,
+    args,
+    kind,
+  } : {
+    contractName : string;
```

Comment:
Create a type for this like you have in other code

Reply:
I will